### PR TITLE
Public token <-> id conversion + Unknown Tokens

### PR DESCRIPTION
**Pull request by pcuenca**
I used an API similar to transformers.
To do (in new PRs, potentially):

- Some tests
- Use unknown token id
- Handle special tokens

cc @ZachNagengast, feel free to review!

### CODE REVIEW

**Review by ZachNagengast — APPROVED** [review]
Great addition - one idea regarding unk token, there could be a parameter in the calling interface for a default token/vocab to use if it's not found in the dictionaries, or these functions could even return an optional

File: Sources/Tokenizers/BPETokenizer.swift [hunk 1]
```diff
@@ -38,13 +38,20 @@ class BPETokenizer: Tokenizer {
...
+    
     private let preTokenizer: PreTokenizer?
     private let normalizer: Normalizer?
     private let postProcessor: PostProcessor?
     private let decoder: Decoder?
     
     private let cleanUpTokenizationSpaces: Bool
 
+    class func unknownToken(from tokenizerConfig: Config) -> String? {
+        return tokenizerConfig.unkToken?.content?.stringValue ?? tokenizerConfig.unkToken?.stringValue
```
Comment (pcuenca) [inline Sources/Tokenizers/BPETokenizer.swift:52]: The first part is the modern way; the second expression appears in distilgpt2, for example.

File: Sources/Tokenizers/BPETokenizer.swift [hunk 2]
```diff
@@ -38,13 +38,20 @@ class BPETokenizer: Tokenizer {
     private let addedTokens: Set<String>
     private let specialTokens: [String: Int]
     
+    public let unknownToken: String?
+    public let unknownTokenId: Int?
```
Comment (pcuenca) [inline Sources/Tokenizers/BPETokenizer.swift:42]: They can be nil now

**Comment by pcuenca** [comment]
@ZachNagengast I further checked the Python code and brought over the relevant changes:

- The unknown token and unknown token id are now read from the tokenizer configuration.
- They can be `nil`. This happens, for example, in the Falcon tokenizer.

Tests now pass for the various tokenizers we are considering, and comparing against the well-known Python unknown token ids.

**Pull request merged**